code/online_challenges/project_euler/problem_036/problem_036.cpp

```diff
@@ -0,0 +1,24 @@
+#include <bitset>
+#include <iostream>
+#include <string>
+
+bool isPalindrome(const std::string& str);
+
+int main()
+{
+    int sum = 0;
+    
+    for (int i = 1; i < 1000000; ++i)
+    {
+        std::string currentBinaryString = std::bitset<32>(i).to_string();
+        currentBinaryString.erase(0, currentBinaryString.find('1')); // Remove leading zeroes
```

Comment:
You can do this after check, then reduce time.

```
#include <bitset>
#include <iostream>
#include <string>
#include <ctime>

std::string convertBase2(int number);
bool isPalindrome(const std::string& str);

int main()
{
    auto MAX = 1000000;

    int sum = 0;
    double c;
    c = clock();

    for (int i = 1; i < MAX; ++i)
        if (isPalindrome(std::to_string(i)))
        {
            std::string currentBinaryString = std::bitset<32>(i).to_string();
            currentBinaryString.erase(0, currentBinaryString.find('1')); // Remove leading zeroes

            if (isPalindrome(currentBinaryString))
                sum += i;
        }

    std::cout << "1  " << (double)(clock()-c) / CLOCKS_PER_SEC << std::endl;
    std::cout << sum << "\n";




    sum = {};
    c = clock();

    for (int i = 1; i < MAX; ++i)
        if (isPalindrome(std::to_string(i)) &&isPalindrome(convertBase2(i)))
            sum += i;

    std::cout << "2  " << (double)(clock()-c) / CLOCKS_PER_SEC << std::endl;
    std::cout << sum << "\n";
}

std::string convertBase2(int number)
{
    std::string result;

    while (number > 0)
    {
        int currentBit = number % 2;
        int currentQuotient = number / 2;
        result += std::to_string(currentBit);
        number = currentQuotient;
    }

    return result;
}

bool isPalindrome(const std::string& str)
{
    auto res = str == std::string{ str.rbegin(), str.rend() };
    return res;
}
```

Reply:
output:
```
1  0.699453
872187
2  0.596954
872187
```

Reply:
At first, I also code as your latest version, then it scared me. 🤣

Reply:
At this point, I feel that you should leave it as it is, since the difference could be considered negligible. 😐

Reply:
um..., I'm not mean that difference between my code.
I mean this:
```
for (int i = 1; i < MAX; ++i)
        if (isPalindrome(std::to_string(i)))
        {
            std::string currentBinaryString = std::bitset<32>(i).to_string();
            currentBinaryString.erase(0, currentBinaryString.find('1')); // Remove leading zeroes

            if (isPalindrome(currentBinaryString))
                sum += i;
        }
```

and yours
```
for (int i = 1; i < 1000000; ++i)
    {
        std::string currentBinaryString = std::bitset<32>(i).to_string();
        currentBinaryString.erase(0, currentBinaryString.find('1')); // Remove leading zeroes
        if (isPalindrome(std::to_string(i)) && isPalindrome(currentBinaryString))
            sum += i;
    }
```